mempool/clist_mempool.go

```diff
@@ -662,12 +662,69 @@ func (mem *CListMempool) recheckTxs() {
 	mem.proxyAppConn.FlushAsync()
 }
 
+func checkCandidate(remainBytes int64, remainGas int64, localTx, starterTx *mempoolTx) bool {
+	if localTx.gasWanted > remainGas || int64(len(localTx.tx)) > remainBytes {
+		return false
+	}
+	if starterTx != nil {
+		if !bytes.Equal(localTx.tx.Hash(), starterTx.tx.Hash()) {
+			return false
+		}
+		if localTx.priority <= starterTx.priority {
+			return true
+		}
+		return false
+	}
+	return true
+}
+
+func (mem *CListMempool) GetNextTransaction(remainBytes int64, remainGas int64, starter []byte) types.Tx {
+	mem.updateMtx.RLock()
+	defer mem.updateMtx.RUnlock()
+	var s types.Tx = starter
+	StarterElement := mem.txs.Front()
+	var StarterTx *mempoolTx
+	if starter == nil {
+		StarterElement = nil
+	} else {
+		// find out clist element related to Starter
+		for StarterElement != nil && !bytes.Equal(StarterElement.Value.(*mempoolTx).tx.Hash(), s.Hash()) {
+			StarterElement = StarterElement.Next()
+		}
+	}
+	if StarterElement == nil {
+		// starter not provided or not found
+		StarterTx = nil
+	} else {
+		StarterTx = StarterElement.Value.(*mempoolTx)
```

Comment:
we don't want `CElement` to be tightly coupled with `mempoolTx`. can we add a `priority` field directly to `CElement`?